cifar10_cnn/periodic_save.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file utils.hpp
+ * @author Kartik Dutt
+ *
+ * Definition of Periodic Save utility functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_CALLBACKS_PERIODIC_SAVE_HPP
+#define ENSMALLEN_CALLBACKS_PERIODIC_SAVE_HPP
+
+#include <ensmallen.hpp>
+#include <mlpack/core.hpp>
+
+namespace ens {
+
+/**
+ * Saves model being trained periodically.
+ *
+ * @tparam ANNType Type of model which will be used for evaluating metric.
+ */
+template<typename AnnType>
+class PeriodicSave
+{
+ public:
+  /**
+   * Constructor for PeriodicSave class.
+   *
+   * @param network Network type which will be saved periodically.
+   * @param filePath Base path / folder where weights will be saved.
+   * @param modelPrefix Weights will be stored as
+   *    modelPrefix_epoch_loss.bin.
+   * @param period Period after which the model will be saved.
+   * @param silent Boolean to determine whether or not to print saving
+   *    of model.
```

Comment:
```suggestion
   *      modelPrefix_epoch_loss.bin.
   * @param period Period after which the model will be saved.
   * @param silent Boolean to determine whether or not to print saving
   *      of model.
```

Should be tabbed twice (4 spaces).